### ISSUE TITLE
Align with and derive from DCAT

### ISSUE THREAD

**Issue by mih** [psychoinformatics-de/datalad-concepts#25]
For example, we should look into modeling a directory after a DCAT `resource`

DCAT v3 (https://www.w3.org/TR/vocab-dcat-3/) has added critical components that have been missing before. This includes

- checksums
- versioning

**Comment by mih** [comment]
After some more reading, I am trying to pinpoint aspects that would encourage *any* deviation from a straight DCAT-v3-based model.

- a `dcat:Dataset` is a conceptual entity. Only a `dcat:Distribution` is a materialization, and a `dcat:Distribution` represents a whole `dcat:Dataset`. This constraint it too strong. We need to be able to represent distributed datasets, where parts can be spread across several `dcat:DataService`s. A no single `dcat:DataService` provides the full dataset.
- `DCAT` provides little for expressing multi-component datasets, and multi-file datasets are the standard case for us. In order to express them efficiently, we need to add more flexibility. We need
  - the concept of a distribution on the file and on the dataset level
  - we need to be able to space-efficiently describe the evolution of (multi-version) dataset -- essentially following the Git data model of snapshots of a tree of trees

### DCAT extension

I think we need to extend in the following ways:

- Any dataset component (not just the dataset itself) needs to support a `distribution`
- support a potentially leaner way of describing the redundant availability of identical materializations of a dataset(components) from different services/locations. It may be sufficient to allow for a multivalue `access_service` property of `dcat:Distribution`.
- provide additional classes to allow for efficient serialization formats
  - `DatasetPart`: a `dcat:Resource`, but would have the `distribution` property and is the common base class for any component of a dataset (`range` of `has_parts`). With a `name` property that enables placement in the containing entity.
  - `File`: a `DatasetPart`,  it's `distribution` would describe file content and availability.
  - `Directory`: a `DatasetPart`, with a `has_parts` that has `DatasetPart` as a multivalue range.

**Comment by jsheunis** [comment]
FTR, I made notes about the DCPAC (Data Catalog – Provenance – Access Control ) ontology at some stage here: https://github.com/psychoinformatics-de/datalad-schema/issues/11. Reading them again, nothing jumps out to me that says we're missing something, especially since we have now incorporated entities from both DCAT and PROV(-O). But perhaps useful to compare our model against to see if entities from ODRL (https://www.w3.org/ns/odrl/2/) in particular could still be useful to define access permissions, etc.
